concepts/05 UI Components/DataGrid/10 Enhance Performance on Large Datasets/030 Rendering Optimization.md

```diff
@@ -1,7 +1,22 @@
+If you work with a large data set, we recommend that you apply these settings to improve rendering performance:
 
-Use virtual row rendering when [paging is disabled](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging/enabled.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/paging/#enabled') or the [pageSize](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging/pageSize.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/paging/#pageSize') is more than the default value.
+* **Render rows and columns of the current viewport only**             
+The default behavior is to render all loaded rows and columns at once. To render those of the current viewport only, set the [rowRenderingMode](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/scrolling/#rowRenderingMode) and [columnRenderingMode](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/scrolling/#columnRenderingMode) properties to *"virtual"*. You should do this for rows if [paging was disabled](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/paging/#enabled) or the **paging**.[pageSize](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/paging/#pageSize) is more than 20 and for columns if more than 20 are outside the current viewport.
 
-Virtual column rendering can be used when there are more than 20 columns outside the viewport.
+* **Render certain elements after others**           
+To implement this, enable the [renderAsync](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#renderAsync) property. It affects filter row, command columns, and columns with [showEditorAlways](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#showEditorAlways) set to **true**.
 
-Cells with complex content (predefined templates, command columns, editors) can be rendered after cells with simple content. To do that, enable the [renderAsync](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/renderAsync.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#renderAsync') property. In addition, you can enable [the same property](/api-reference/_hidden/GridBaseColumn/renderAsync.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#renderAsync') for individual columns if they use a content-heavy [cellTemplate](/api-reference/_hidden/dxDataGridColumn/cellTemplate.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#cellTemplate').
+* **Render columns with a complex [template](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#cellTemplate) after others**               
+UI component's **renderAsync** property does not affect such columns - enable the **columns[]**.[renderAsync](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#renderAsync) property for them.
```

Comment:
1. Looks like these two items cover the same ground and should be collected together.
2. What are the drawbacks of enabling renderAsync? Based on the description alone, async rendering looks like a cool feature that should be enabled by default. However, that isn't so, and the docs doesn't explain why.

Reply:
renderAsync was introduced after the synchronous render => enabling this option would result in a BC.